monitoring/bin/deploy_monitoring_cluster.sh

```diff
@@ -229,6 +220,24 @@ else
   log_debug "PrometheusRule $MON_NS/v4m-kubernetes-apps does not exist"
 fi
 
+# Elasticsearch Datasource for Grafana
+# Moved down to make sure Grafana pods exist
+ELASTICSEARCH_DATASOURCE="${ELASTICSEARCH_DATASOURCE:-false}"
+if [ "$ELASTICSEARCH_DATASOURCE" == "true" ]; then
+  set +e
+  monitoring/bin/create_elasticsearch_datasource.sh
+
+  if (( $? == 1 )); then
+    log_warn "Unable to configure the Elasticsearch data source at this time."
+    log_warn "Please address the errors and re-run the follow command:"
```

Comment:
I'm not comfortable with a WARNing message talking about "errors"...if there were errors, shouldn't it be an ERROR message?  But I agree that a failure to create the datasource should NOT cause an ERROR here.  So, let's tweak the wording of the message.  It would be nice if the called script had a flag that told it to fail silently or, at least, without an ERROR message.  But that's not there and isn't absolutely needed. Maybe reword things as "You may configure the datasource later, after addressing any issues noted above, by running the following command:"

Reply:
So the problem here is that I'm not aware of a way to make it choose a log level for the error depending on whether it's run independently or not.
When run on its own, it IS supposed to error out.  When run within the deployment scripts, it should still show the regular error (so that people know how to fix it), but the WARNING assures that the script will still run.  I think it's important that, although an error popped up, they can re-run the script and try again after it has been addressed.

Reply:
Ok, so there currently isn't a mechanism that would allow us to have create_elasticsearch_datasource.sh generate a WARNing rather than an ERROR message when it fails.  While that would be nice to have, it isn't critical.  The current wording here makes the ERROR messages seem more significant than they are.  And, tells the user they need to do something.  The reality is the ERRORs aren't a big deal (which is why we are continuing) and the user only has to do something if they want to make the log messages available in Grafana.  I believe my suggested rewording conveys that information.  If you feel it needs to be tweaked, feel free.